### ISSUE TITLE
How to change the highlight of the cursorline inside the preview window?

### ISSUE THREAD

**Issue by skamsie** [junegunn/fzf.vim#1179]
- [x] I have fzf 0.23.0 or above
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [x] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

Is there any way of configuring the highlight of the cursorline inside the `Rg` or `Ag` preview? I have read [this issue](https://github.com/junegunn/fzf.vim/issues/969) and installed `bat` and set `BAT_THEME` to `export BAT_THEME="ansi-dark"` which should use the default terminal colors. The colors are ok, the problem is the color of the selected text is the same with the text color. Is there a way to change it? or at least set it to underline?

<img width="1440" alt="Screenshot 2020-11-16 at 20 26 59" src="https://user-images.githubusercontent.com/7014744/99300027-5c886400-284c-11eb-932c-340180ea517d.png">

**Comment by skamsie** [comment]
If I uninstall `bat` it looks like this (which is event better in terms of colors as it seems to trully use my terminal colors). But the problem is still the cursorline... It is now a rainbow which is still something that I would like to change 🌈.

<img width="1438" alt="Screenshot 2020-11-16 at 21 51 25" src="https://user-images.githubusercontent.com/7014744/99306893-159f6c00-2856-11eb-9216-69e342708968.png">

**Comment by jsit** [comment]
I believe this is determined by your bat theme: https://github.com/sharkdp/bat/issues/1457

I'm seeing the same issue while using the `ansi-light` theme. You may have luck using a different theme. Also bat will soon drop `ansi-dark` and `ansi-light` in favor of a single `ansi` theme: https://github.com/sharkdp/bat/blob/master/CHANGELOG.md#new-themes

**Comment by iamvoidcoder** [comment]
Skamsie, have you got to the bottom of highlighted line inverted colors issue? I have been messing with this for a couple of evenings now and can't figure what is going on. This is driving me insane. Struggling with color mash of matched lines, especially the long ones.   

I don't think this has anything to do with the Bat theme. At least highlighted line is colored fine when running FZF as a standalone tool with the same Bat theme. The issue only occurs when using FZF.vim plugin within Vim:

E.g. highlighted line in preview window in standalone fzf: 

<img width="614" alt="Screenshot 2021-04-10 at 02 10 52" src="https://user-images.githubusercontent.com/81540569/114254102-63ec0d80-99a5-11eb-876f-4bfbcdb761aa.png">

 
The same highlighted line in fzf vim: 

<img width="654" alt="Screenshot 2021-04-10 at 02 08 21" src="https://user-images.githubusercontent.com/81540569/114254113-6d757580-99a5-11eb-938b-e3fb109973ee.png">

**Comment by skamsie** [comment]
@iamvoidcoder Nope, just got used to it after a while... I uninstalled `Bat` because I have no use for it and it does not solve the problem anyway.

**Comment by junegunn** [comment]
As @jsit pointed out, the highlighting is done by bat itself with its `--highlight-line` option.

```sh
export BAT_THEME="ansi-dark"
bat README.md --highlight-line 1
```

You might want to test with different themes (`bat --list-themes`).

@iamvoidcoder Try running the `bat .. --highlight-line ..` command on your normal terminal and on the terminal buffer inside Vim (`:terminal`) and see if they show different results.

**Comment by iamvoidcoder** [comment]
I realise it could be Bat itself. Yet I spent quite a bit of time trying to debug this issue through both Bat & Syntect code and didn't get anywhere.

If FZF is run as an external standalone tool in either normal terminal (MacOS standard Terminal or iTerm2) or Neovim built in terminal then Bat line highlight in FZF preview is fine. Line highlight is correct as per current Bat color theme.  

However FZF.vim plugin preview will show inverted colors for the highlighted line. This is only to do with the highlighted line, other lines in preview are colored as they should and look identical to standalone FZF preview. 

Bat theme doesn't change anything, I tried them all. It is the same result with all Bat themes. It appears with FZF.vim the highlighted line colors are as follows:

- All text foreground colors become background colors
- Foreground color becomes while #FFFFFF

Here are three screens for running FZF in external terminal, nvim internal terminal and nvim as a plugin:

FZF standalone, external terminal, export BAT_THEME="default"

<img width="604" alt="Bat Default Ext Terminal" src="https://user-images.githubusercontent.com/81540569/114267629-bef92100-99f4-11eb-9d5b-1a0156843b77.png">


FZF standalone, nvim :terminal, export BAT_THEME="default"

<img width="590" alt="Bat Default Nvim Terminal" src="https://user-images.githubusercontent.com/81540569/114267635-cb7d7980-99f4-11eb-9e2c-45314dc8b7fa.png">


FZF.vim plugin, let $BAT_THEME='default'

<img width="606" alt="Bat Default Nvim Plugin" src="https://user-images.githubusercontent.com/81540569/114267640-d0422d80-99f4-11eb-9d30-678327839f69.png">

**Comment by junegunn** [comment]
Do you have the latest version of fzf.vim?

**Comment by iamvoidcoder** [comment]
I'm pulling it via vim-plug. PlugUpdate saying fzf.vim is up to date. Also checked the repo just in case, appears to be top of the tree #caf7754:  

```
$ cd /Users/vcd/.config/nvim/autoload/plugged/fzf.vim
$ git log -1 --format="%h"
caf7754
```

fzf is also latest:

```
$ fzf --version
0.27.0 (brew)
```

Bat also latest:

```
$ bat --version
bat 0.18.0
```

Neovim:

```
$ nvim --version
NVIM v0.4.4
```

I have just tried on Ubuntu box, just in case. Same issue exactly. 

Are you saying the highlighted line works ok in fzf.vim preview for you? No inverted foreground->background rainbow colors?

**Comment by iamvoidcoder** [comment]
Interesting, I have just discovered the issue is only present if I overwrite FZF_PREVIEW_COMMAND environment variable in my init.vim.

`let $FZF_PREVIEW_COMMAND="bat --style=numbers --color=always {}"`

If I comment this line out the highlighted line colors are back to normal. Which is rather strange because same exactly preview settings with standalone fzf tool work fine. E.g. this shows highlighted line just fine:

`fzf --preview 'bat --style=numbers --color=always --highlight-line 1:1 {}'`

Any ideas please?

**Comment by junegunn** [comment]
If you use that variable, [the scipt](https://github.com/junegunn/fzf.vim/blob/master/bin/preview.sh) does the highlighting, so you get the different result. `FZF_PREVIEW_COMMAND` is an arbitrary command, and it's not clear to the script that it can inject `--highlight-line` option for bat which may or may not be compatible with the custom command.

```vim
let $FZF_PREVIEW_COMMAND = "non-bat-program --with-its-own-options {}"

" We can't just append --highlight-line at the end of the command even if it starts with "bat"
let $FZF_PREVIEW_COMMAND = "bat {} 2> /dev/null || alternative-bat {}"
```

Because of the limitation, the variable is [undocumented on purpose](https://github.com/junegunn/fzf.vim/pull/707#issuecomment-437273217). Where did you find it?

**Comment by iamvoidcoder** [comment]
Honestly I don't quite remember where I got it from. I was googling around for a way to overwrite default fzf.vim bat command for a totally different reason when I came across it. 

The issue is that Bat is not using full 24-bit color when run in context on nvim fzf.vim (MacOS, iTerm2). The colors appears quite a bit off. I attempted to dig this a bit further and I think the issues is that $COLORTERM=truecolor variable is getting lost somehow by the time nvim->fzf->bat gets invoked. So the only solution I found to have 24-bit color in fzf.vim bat preview was:

`let $FZF_PREVIEW_COMMAND="COLORTERM=truecolor bat --style=numbers --color=always {}" `

**Comment by junegunn** [comment]
Have you tried setting `export COLORTERM=truecolor` in your shell configuration file or `let $COLORTERM = 'truecolor'` in your Vim configuration file? I don't see how it wouldn't work.

**Comment by iamvoidcoder** [comment]
I don't need to add `export COLORTERM=truecolor`  because iTerm2 is setting this variable out of the box. It is indeed set in nvim's context, there is nothing else I need to do in either shell config file nor in my init.vim:

```
:echo $COLORTERM
truecolor
```

Yet it looks like somehow by the time Bat gets called $COLORTERM gets lost. I can't explain what is going on but it is evident that adding `COLORTERM=truecolor` to FZF_PREVIEW_COMMAND in my init.vim is fixing Bat 24-bit syntax coloring.

let $FZF_PREVIEW_COMMAND="COLORTERM=truecolor bat --style=numbers --color=always {}"

Removing `COLORTERM=truecolor` from FZF_PREVIEW_COMMAND or removing FZF_PREVIEW_COMMAND from init.vim altogether is resulting in 8-bit color in Bat preview. You can still recognise the selected Bat theme but the colors are quite a bit off. Like e.g. light grey becomes white, dark grey becomes black etc.

A mystery.

**Comment by minhdanh** [comment]
Hi @iamvoidcoder, it's rather unrelated to the issue, but can I ask what colorscheme/theme are you using for vim in this comment: https://github.com/junegunn/fzf.vim/issues/1179#issuecomment-817050449 ? I really love it 😄
